javascript/javascript1/week4/script.js

```diff
@@ -0,0 +1,128 @@
+let userName;
+
+const toDoList = [];
+
+const userNameIntro = ['Hello', 'my', 'name', 'is'];
+const userNameRequest = ['What', 'is', 'my', 'name'];
+const userAddToDos = ['Add', 'to', 'my', 'todo'];
+const userRemovedFromToDos = ['Remove', 'from', 'my', 'todo'];
+const userRequestToDo = ['What', 'is', 'on', 'my', 'todo'];
+const userRequestDay = ['What', 'day', 'is', 'it', 'today?'];
+const userRequestCalculation = ['What', 'is'];
+const userRequestSetTimer = ['Set', 'a', 'timer', 'for', 'minutes'];
+const userRequestMood = ['How', 'is', 'your', 'mood', 'today?'];
+
+const findDifferenceBetweenReqAndRes = (request, possibleReq) => {
+  return request.filter((item) => !possibleReq.includes(item)).join(' ');
+};
+
+const checkForSimilarity = (userRequest, possibleRequest) => {
+  return possibleRequest.every((item) => userRequest.includes(item));
+};
+
+const getReply = (command) => {
+  const userCommand = command.split(' ');
+
+  if (checkForSimilarity(userCommand, userNameIntro)) {
+    userName = findDifferenceBetweenReqAndRes(userCommand, userNameIntro);
+
+    if (userName[0] === '' || userName.length === 0) {
```

Comment:
Check ```getReply("Hello my name is          ")``` and press space several types before typing `")` at the end

Reply:
`if (!username) ` also covers empty strings and undefined, null etc

Reply:
Fixed 👍